### ISSUE TITLE
[Bug] areaStyle origin start/end not work propley when yAxis.type='value'

### ISSUE THREAD

**Issue by messaoudi-mounir** [apache/echarts#18628]
### Version

5.4.2

### Link to Minimal Reproduction

https://echarts.apache.org/examples/en/editor.html?c=line-simple&code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMmurGAJ4gCmRA5AG4CGANgK5W0A0JAvt-uRz4iqUmUo1YDFuy4l0AWwjQCAJgAMm_qQWMMajet7a8VAE4QqhWAG15xMegAmjMIyI2bh9ZwCsAXU5bVQBGLQA2QODDTjD_QPt0Cmo6ZmUObTE8BWBgMAALIjAzdkzSKmgAY2AnGlFHdH0QsrFyAiMGvkTYRjMqRgBlCmY67vRgCwBzZQJaCqdabp57ZfR_FB4AbiA

### Steps to Reproduce

1. create a line chart with 2 axis with type values

`option = {
   xAxis: {
     type: 'value',
   },
   yAxis: {
     type: 'value',
     min:2000,
     max:2200
   },
   series: [
     {
       data: [[2000,5], [2100,6], [2200,10]],
       type: 'line',
       smooth: true,
       encode:{
         x:1,
         y:0
       },
       areaStyle:{
         origin:'start'
       }
     }
   ]
 };
`

### Current Behavior

![image](https://github.com/apache/echarts/assets/222980/55f3dfd9-63be-4b8b-aee9-2cd15c881a3b)

### Expected Behavior

![image](https://github.com/apache/echarts/assets/222980/d34074cc-05d0-47d2-ab63-9f4519d5c2f2)

if you change yAxis.type to 'time' or 'category', the area style is filled correctly.

https://echarts.apache.org/examples/en/editor.html?c=line-simple&code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMmurGAJ4gCmRA5AG4CGANgK5W0A0JAvt-uRz4iqUmUo1YtSAFsO_UjIjQCAJgAMmhehmMMajet4K8VAE4QqhWAG0S6UWNgATRmEZEbNw-s4BWAF1OW1UARi0ANiCQw05wgKD7UgpqOmZleST0PBlgYDAACyIwM3ZtUipoAGNgZxpHJ0wCUPKxcgIjRr4s2EYzKkYAZQpmep70YAsAc2UCWjx3MzBaHp4ktfQAlB4AbiA

### Environment

```markdown
- OS: Windows 10
- Browser: Firefox/Chrome
- Framework: echarts.apache.org/examples
```

**Comment by helgasoft** [comment]
yes, unfortunately the [documentation](https://echarts.apache.org/en/option.html#series-line.areaStyle.origin) does not say from **what axis** the 'start' or 'end' are measured. As you pointed out, it switches depending on axis type.
Possible workaround - add a hidden yAxis type 'time' [Demo Code](https://echarts.apache.org/examples/en/editor.html?c=line-simple&code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMrAL4A0asAnjvkQNqXpKxhUgCmRA5AG4CGAGwCuHbmVgBbCNAIAmAAxKJk_hnmKFpCumat2XWN0iSxEvAAtgAdyIAzIXg6lKAXR2wnAJwgdCsRl1kJl0AE34wfgZ6TQUyAFZ3ALkARmUANiSYzTI013cQ9DZOHkEZM0LPSWBgMAsiMC9RDyCOaABjYFCuFnUUiSoCLXJKmlw8AElobvVYfsr-Lw5-AGU2QR7K9GAfAHMZAm48SK8wbkqSEMv0VxQSAG4gA).
![image](https://github.com/apache/echarts/assets/13038071/68f5a089-1926-4041-acd6-2e77734d847e)
NB: please close issue if problem solved.
